stdlib/tests/collections/smt.rs

```diff
@@ -0,0 +1,124 @@
+use crate::build_test;
+use test_utils::{
+    crypto::get_remaining_key, proptest::prelude::*, seeded_word, EmptySubtreeRoots, Felt,
+    IntoBytes, MerkleStore, NodeIndex, Rpo256, StarkField, Word, WORD_SIZE, ZERO,
+};
+
+#[test]
+fn foo() {
+    let seed = [0xff; 32];
+    let a: Word = test_utils::rand::prng_array(seed);
+    let b: Word = test_utils::rand::prng_array(seed);
+    assert_eq!(a, b);
+}
+
+proptest! {
+    #[test]
+    fn arbitrary_smtget_opens_correctly(
+        ref seed in any::<[u8; 32]>()
+    ) {
+        // setup the test
+        let empty = EmptySubtreeRoots::empty_hashes(64);
+        let initial_root = Word::from(empty[0]);
+        let mut seed = *seed;
+        let key = seeded_word(&mut seed);
+        let value = seeded_word(&mut seed);
+
+        // check empty tree
+        let store = MerkleStore::new();
+        let advice_map = [];
+        let expected = [ZERO; WORD_SIZE];
+        assert_case_arbitrary_smtget_opens_correctly(
+            key,
+            expected,
+            initial_root,
+            store,
+            &advice_map
+        );
+
+        // check included leaves for all tiers
+        for depth in [16, 32, 48, 64] {
+            // compute the index and node value
+            let index = key[3].as_int();
+            let index = index >> (64 - depth);
+            let index = NodeIndex::new(depth as u8, index).unwrap();
+            let remaining = get_remaining_key(key, depth as u8);
+            let node =
+                Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(depth)).into();
+
+            // setup the store and run the test
+            let mut store = MerkleStore::new();
+            let root = store.set_node(initial_root, index, node).unwrap().root;
+            let advice_map = [(node.into_bytes(), value.to_vec())];
+            assert_case_arbitrary_smtget_opens_correctly(
+                key,
+                value,
+                root,
+                store,
+                &advice_map
+            );
+        }
```

Comment:
I'd probably add another test similar to this one but initialize the tree with two nodes sharing the same 16, or 32, or 48, or 64-bit prefix and then do a lookup on one of these nodes.

If doing this via proptest becomes complicate, I think doing a simple test is fine too.

Reply:
Broke the test into 3 variants without proptest